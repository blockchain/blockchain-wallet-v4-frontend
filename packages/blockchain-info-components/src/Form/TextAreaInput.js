import React from 'react'
import styled from 'styled-components'

const BaseTextAreaInput = styled.textarea`
  display: block;
  width: 100%;
  padding: 6px 12px;
  box-sizing: border-box;
  font-size: 14px;
  color: ${props => props.theme['gray-5']};
  background-color: ${props => props.theme['white']};
  background-image: none;
  outline-width: 0;
  user-select: text;
  font-family: 'Montserrat', Helvetica, sans-serif;
  resize: ${props => (props.resize ? 'initial' : 'none')};
  border: 1px solid ${props => props.theme['gray-2']};

  &::-webkit-input-placeholder {
    color: ${props => props.theme['gray-2']};
  }
`

const selectBorderColor = state => {
  switch (state) {
    case 'initial':
      return 'gray-2'
    case 'invalid':
      return 'error'
    case 'valid':
      return 'success'
    default:
      return 'gray-2'
  }
}

const TextAreaInput = props => {
  const { errorState, ...rest } = props
  const borderColor = selectBorderColor(errorState)
  return <BaseTextAreaInput borderColor={borderColor} {...rest} />
}

TextAreaInput.defaultProps = {
  resize: true
}

export default TextAreaInput
