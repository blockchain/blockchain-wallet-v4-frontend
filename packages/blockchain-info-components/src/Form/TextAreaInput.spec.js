import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import TextAreaInput from './TextAreaInput'

describe('TextAreaInput component', () => {
  it('default renders correctly', () => {
    const component = shallow(<TextAreaInput />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
