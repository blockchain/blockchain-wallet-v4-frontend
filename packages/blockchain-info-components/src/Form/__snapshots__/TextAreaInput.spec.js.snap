// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaInput component default renders correctly 1`] = `
<TextAreaInput__BaseTextAreaInput
  borderColor="red"
  resize={true}
/>
`;
